packages/bot-web-ui/src/pages/dashboard/announcements/announcement-dialog.tsx

```diff
@@ -1,16 +1,16 @@
 import React from 'react';
-import { Dialog, Icon, Text } from '@deriv/components';
-import { Localize } from '@deriv/translations';
+import { Dialog, Text } from '@deriv/components';
 import { LabelPairedCheckCaptionFillIcon } from '@deriv/quill-icons';
 import './announcement-dialog.scss';
 import { TAnnounce, TContentItem } from './config';
+import { IconAnnounceModal } from './announcement-components';
 
 type TAccumulatorAnnouncementDialog = {
     announcement: TAnnounce;
     is_announce_dialog_open: boolean;
     setIsAnnounceDialogOpen: (is_announce_dialog_open: boolean) => void;
     handleOnConfirm: () => void;
-    handleOnCancel: () => void;
+    handleOnCancel: (() => void) | null;
```

Comment:
@maryia-matskevich-deriv will it be ideal to make this property optional instead of accepting null additionally?

Reply:
@shafin-deriv, is it okay to use `undefined` instead of `null` here?              
<AnnouncementDialog
                    announcement={selected_announcement.announcement}
                    is_announce_dialog_open={is_announce_dialog_open}
                    setIsAnnounceDialogOpen={setIsAnnounceDialogOpen}
                    **handleOnCancel={!selected_announcement?.should_not_be_cancel ? handleOnCancel : null}**
                    handleOnConfirm={handleOnConfirm}
                    is_tablet={is_tablet}

Reply:
It is okay but let's stick to null in this case.